core/os/os.cpp

```diff
@@ -494,7 +494,13 @@ int OS::get_power_percent_left() {
 	return -1;
 }
 
+void *OS::get_stack_bottom() const {
+	return _stack_bottom;
+}
+
 OS::OS() {
+	void *volatile stack_bottom;
```

Comment:
Are you sure in main()s there is, and will be no pointer before OS() constructor?

Maybe it would be better to get that address as first line in main() and then pass it to OS() constructor?